lib/omnisharp-atom/features/buffer-updater.ts

```diff
@@ -0,0 +1,106 @@
+import Omni = require('../../omni-sharp-server/omni');
+import OmniSharpAtom = require('../omnisharp-atom');
+import OmniSharpClient = require('omnisharp-client');
+import _ = require('lodash');
+import EventKit = require('event-kit');
+import ClientManager = require('../../omni-sharp-server/client-manager');
+
+class BufferUpdater {
+    private disposables: EventKit.CompositeDisposable;
+    private editor: Atom.TextEditor;
+    private emitter: EventKit.Emitter;
+
+    constructor(editor: Atom.TextEditor) {
+        this.disposables = new EventKit.CompositeDisposable();
+        this.emitter = new EventKit.Emitter();
+        this.editor = editor;
+
+        var buffer = editor.getBuffer();
+
+        this.disposables.add(buffer.onDidSave(() => this.updateBuffer(editor)));
+        this.disposables.add(buffer.onDidReload(() => this.updateBuffer(editor)));
+        this.disposables.add(buffer.onDidDestroy(() => this.dispose()));
+
+        this.disposables.add(buffer.onDidChange(event =>  this.changeBuffer(editor, event)));
+
+        this.disposables.add(atom.workspace.observeActivePaneItem((pane: Atom.Pane) => {
+            var activeEditor = atom.workspace.getActiveTextEditor();
+            if (activeEditor != null && this.editor.id == activeEditor.id)
```

Comment:
I got this from linter where they also compare id's for the editor. This affects Typescript not sure how to deal with it. /cc @david-driscoll